Automattic/jetpack
# Enable ESC and clicking background to close full screen dialog

Fixes #7904

#### Changes proposed in this Pull Request:

* Clicking the background or pressing the ESC key should close full screen dialogs
* Move close button into the dialog body.

<img width="614" alt="close-button-moved" src="https://user-images.githubusercontent.com/51896/35651134-67a95c32-0693-11e8-9820-19b589f2c8bf.png">

#### Testing instructions:

* Trigger a full-screen dialog. This code snippet in a micro-plugin will do the trick for forcing an upgrade prompt on any of the Jetpack dash/settings pages:

```php
add_action( 'plugins_loaded', function() {
    Jetpack::state( 'message', 'modules_activated' );
});
```

* Click the background, the dialog should be closed
* Click the foreground (main part of dialog or picture above it) - the dialog should NOT be closed
* Re-open the dialog (e.g. by re-loading). Hit the ESC key. The dialog should be closed.

This also moves the close button into the top-right corner of the dialog, and includes an updated gridicons file which includes the circular close icon. Other uses of gridicons should be re-tested to ensure I didn't break anything.


#### Proposed changelog entry for your changes:

Better usability for full-screen dialogs.
